include/lsst/afw/image/VisitInfo.h

```diff
@@ -86,14 +86,15 @@ class VisitInfo : public table::io::PersistableFacade<VisitInfo>, public typehan
      * @param[in] observatory  observatory longitude, latitude and altitude
      * @param[in] weather  basic weather information for computing air mass
      * @param[in] instrumentLabel  The short name of the instrument that took this data (e.g. "HSC")
+     * @param[in] id  The identifier of this full focal plane data.
      */
     explicit VisitInfo(table::RecordId exposureId, double exposureTime, double darkTime,
                        daf::base::DateTime const &date, double ut1, lsst::geom::Angle const &era,
                        lsst::geom::SpherePoint const &boresightRaDec,
                        lsst::geom::SpherePoint const &boresightAzAlt, double boresightAirmass,
                        lsst::geom::Angle const &boresightRotAngle, RotType const &rotType,
                        coord::Observatory const &observatory, coord::Weather const &weather,
-                       std::string const &instrumentLabel)
+                       std::string const &instrumentLabel, table::RecordId const &id)
```

Comment:
From butler this is a python integer (64-bit I think in registry)  so I'm not entirely why this is a RecordId. Does that mean something special such that you could use it as a primary key in an afw table or something?

Reply:
`RecordId` is a typedef to a signed 64-bit int.  Don't know if it's better or worse to use the typedef here; it guarantees consistency with the ID columns in afw.table that sometimes hold these values, but it's also not a desirable dependency (even if it's not the only occurrence of this dependency).

Reply:
I'm using the same type as `exposureId`, since this effectively replaces that. `table::RecordId` is int64.